### DISCUSSION TITLE
3D field with particle orbits

### DISCUSSION THREAD

**Discussion by fwessel** [PlasmaPy/PlasmaPy#1240] [Simulations]
Any suggestions from the group on how to plot the magnetic field lines in a 3D volume, with particle gyro-orbits superimposed? All insights would be welcome. Thanks.

**Comment by StanczakDominik** [comment]
For plotting 3D stuff in Python, I generally use [Pyvista](https://docs.pyvista.org/) nowadays. Could you say something more about what actual data you've got for plotting? Is it grid data, trajectories?

**Reply by fwessel** [reply]
Thanks Dominik, 
I got the fields working in a Juypter notebook, with a little struggle. I will give Pyvista a look. 
Yes, the overlay are trajectory data, computed for the field at each time step. 
ATM it's still a work in progress, especially trying to match everything with the integrator, and one that conserves energy.
